templates/vue/src/store/getters.js

```diff
@@ -93,27 +93,37 @@ export function isVisible(state, { getNode, hasMultiContentAncestor }) {
   }
 }
 
-export function getActivities(state) {
-  return (options = {}) => {
-    const { exclude = [] } = options
-    return Object.values(state.nodes)
-      .filter(node => !exclude.includes(node.id) && Boolean(node.quiz))
-      .flatMap(node => node.quiz)
-  }
-}
-
 export function getQuestion(state) {
   return id => {
     const node = Object.values(state.nodes)
-      .filter(node => node.quiz)
-      .find(node => node.quiz.find(q => q.id == id))
+      .filter(node => node.typeData.activity)
+      .find(node => node.typeData.activity.questions.find(q => q.id == id))
     if (node) {
-      return node.quiz.find(q => q.id == id)
+      return node.typeData.activity.questions.find(q => q.id == id)
     }
     return null
   }
 }
 
+export function getAnswers(state) {
+  return (nodeId, questionId) => {
+    if (state.userAnswers.hasOwnProperty(nodeId)) {
+      if (state.userAnswers[nodeId].hasOwnProperty("activity")) {
+        if (state.userAnswers[nodeId].activity.hasOwnProperty(questionId)) {
+          if (
+            typeof state.userAnswers[nodeId].activity[questionId].answers !==
```

Comment:
Can you not just use `!state.userAnswers[nodeId].activity[questionId].answers` here?

Reply:
Hi @wynnset, I actually do need to check if each property has been initialized yet(nested if), because if this is the first time a user opens that activity and there is no userAnswers for that activity yet, checking with only that line will outputs console errors
<img width="523" alt="Screen Shot 2021-06-17 at 6 13 34 PM" src="https://user-images.githubusercontent.com/30706624/122490887-ceb65800-cf97-11eb-8660-4a0c5bf8f382.png">

Reply:
Hey @harry02031 , I was talking about only the one line that I commented on. But you can also get rid of the other if statements by using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

Reply:
Hi @wynnset, that string checking is needed because if a user has never done that activity before, state.userAnswers[nodeId].activity[questionId].answers would always start off as a empty string
<img width="536" alt="Screen Shot 2021-06-17 at 11 15 28 PM" src="https://user-images.githubusercontent.com/30706624/122826734-594ccf00-d298-11eb-8c36-a41ac560afe9.png">

Reply:
@harry02031 but if it's an empty string, why do you want to return it? You should just add an else statement and return `{}`, no?